internal/component/loki/source/file/tailer.go

```diff
@@ -44,15 +44,17 @@ type tailer struct {
 	posAndSizeMtx sync.Mutex
 
 	running *atomic.Bool
+	stopped *atomic.Bool
```

Comment:
This struct has so many fields for controlling its lifetime. Could it be simplified so that we don't need all of them? If the code is simpler then there's less chance for bugs too...

Reply:
We could merged these to into an `atomic.Int32` and call it state. Then have something like

```
const (
  stateRunning int32 = 1
  stateStopped int32 = 2
)
```

And transition the state instead if you think that would be better

Reply:
Check [a2a5f5e](https://github.com/grafana/alloy/pull/3452/commits/a2a5f5e1a161bfbf61ce78cfc9ca2aade7053038). I changed to use what I proposed. 

We now have tree states, idle, running and stopped. And you can only transition from idle->running->stopped.

With this I am pretty sure we don't need the initOnce function, will check that. And now a task can only be triggered once so this is more like this component behaved before https://github.com/grafana/alloy/pull/2428 but we still "retry" for missing files.